### PR TITLE
Add coverity annotation to suppress false positive in rbtree.c

### PR DESCRIPTION
**Pull request by MaskRay**
The root is black, when q is red, it cannot be the root and its parent exists

### CODE REVIEW

File: libr/util/rbtree.c [hunk 1]
```diff
@@ -190,7 +190,9 @@ R_API void r_rbtree_aug_insert(RBNode **root, void *data, RBNode *node, RBCompar
 				q->red = true;
 			}
 		}
-		if (q->red && p && p->red) {
+		// coverity[var_deref_op] false positive
+		// The root is black, when q is red, it cannot be the root and its parent (p) must exist (not NULL).
+		if (q->red && p->red) {
```
Comment (radare) [inline libr/util/rbtree.c]: adding a comment doesnt saves you from a null deref

**Comment by condret** [comment]
don't assume things, bc of 2 reasons:
1. We're not smart enough to verify if you are smart enough to proove your assumptions or even to verify your proof.
2. Newbie coders could see that and learn, that it is ok to remove the checks, if they think they are pointless.

Beeing more defensive does not hurt

**Comment by MaskRay** [comment]
@condret  The check is self deceiving. It does not protect you. If the root was red, there would many ways it would break (including segfault).

**Comment by MaskRay** [comment]
> Newbie coders could see that and learn, that it is ok to remove the checks, if they think they are pointless.

That was why I used a Coverity annotation. An annotation / comment also conveys the idea that the author has carefully thought about this and guides readers to the correct redirection.
the non-existence of null deref check helps ppl understand the code.
if the check was used, i would think when it would be NULL and whether the code had a malformed intermediate stage.

**Pull request closed without merging**